### PR TITLE
archive: add v3.0 and v7.2 docs

### PR DESCRIPTION
**Pull request by qiancai**
1. Add v3.0 and v7.2 to the home page of the Archive Docs website.
2. Update the version selection list on the Archive Docs website.

### CODE REVIEW

File: markdown-pages/en/tidb/_docHome.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ hide_leftNav: true
 
 | Release    | Archived documentation                                         | Archive date     |
 | ---------- | -------------------------------------------------------------- | ---------------- |
+| v7.2 (DMR) | [TiDB v7.2 documentation](https://docs-archive.pingcap.com/tidb/v7.2/) | October 20, 2023 |
+| v3.0       | [TiDB v3.0 documentation](https://docs-archive.pingcap.com/tidb/v3.0/) | October 20, 2023 |
```
Comment (Yuiham) [inline markdown-pages/en/tidb/_docHome.md:24]: Move this line below `v3.1`?

File: markdown-pages/zh/tidb/_docHome.md [hunk 2]
```diff
@@ -20,6 +20,8 @@ hide_leftNav: true
 
 | 版本        | 归档文档链接                                             | 归档日期 |
 | ---------- | ------------------------------------------------------ | ----------------- |
+| v7.2 (DMR) | [TiDB v7.2 文档](https://docs-archive.pingcap.com/zh/tidb/v7.2) | 2023 年 10 月 20 日 |
+| v3.0       | [TiDB v3.0 文档](https://docs-archive.pingcap.com/zh/tidb/v3.0) | 2023 年 10 月 20 日 |
```
Comment (Yuiham) [inline markdown-pages/zh/tidb/_docHome.md:24]: The same

Comment (qiancai) [reply markdown-pages/zh/tidb/_docHome.md:31]: done in https://github.com/pingcap/docs-staging/pull/54/commits/dbd0bc1926a45642862f3389958e045ec4ae188c.

**Review by Yuiham — APPROVED** [review]
LGTM

**Pull request merged**